test/new-e2e/examples/customenv_with_docker_app_test.go

```diff
@@ -0,0 +1,186 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package examples
+
+import (
+	_ "embed"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner/parameters"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DataDog/test-infra-definitions/components/docker"
+	"github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/resources/aws"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/fakeintake"
+
+	"github.com/DataDog/test-infra-definitions/components/datadog/agent"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type vmPlusDockerEnv struct {
+	RemoteHost *components.RemoteHost
+	Agent      *components.RemoteHostAgent
+	Fakeintake *components.FakeIntake
+	Docker     *client.Docker
+
+	// additional resources
+	remoteHostLogsDir string
+}
+
+var _ e2e.Initializable = &vmPlusDockerEnv{}
+
+// Init initializes the environment
+func (e *vmPlusDockerEnv) Init(ctx e2e.Context) error {
+	privateKeyPath, err := runner.GetProfile().ParamStore().GetWithDefault(parameters.PrivateKeyPath, "")
+	if err != nil {
+		return err
+	}
+
+	// initialize docker client
+	e.Docker, err = client.NewDocker(ctx.T(), e.RemoteHost.HostOutput, privateKeyPath)
+	if err != nil {
+		return err
+	}
+
+	// initialize agent host
+	agent, err := client.NewHostAgentClient(ctx.T(), e.RemoteHost, true)
+	if err != nil {
+		return err
+	}
+	e.Agent.Client = agent
+
+	return nil
+}
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
we might get rid of this